command line option '-std=gnu++11' is valid for C++/ObjC++ but not for C

In [`M2/configure.ac`](https://github.com/Macaulay2/M2/blob/master/M2/configure.ac#L1279), we have the following:

     FFLAS_FFPACK_CPPFLAGS=`fflas-ffpack-config --cflags 2>&1`
     if test $? = 0
     then AC_MSG_RESULT(found)
          M2_CPPFLAGS="$M2_CPPFLAGS $FFLAS_FFPACK_CPPFLAGS"

So if we're using the system fflas-ffpack, whatever flags `fflas-ffpack-config` spits out will be added to `M2_CPPFLAGS`.  For example:

    root@cayke:/tmp/buildd/macaulay2-1.9.2-10-g06c3063# fflas-ffpack-config --cflags
    -I/usr/include -D__FFLASFFPACK_HAVE_CBLAS -g -O2 -fdebug-prefix-map=/build/fflas-ffpack-dZEHwx/fflas-ffpack-2.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -fno-strict-aliasing -fabi-version=6 -std=gnu++11 -fopenmp

In particular, we have `-std=gnu++11`.

Later on in the build, we try building C source with this C++-specific flag in CPPFLAGS, giving us the following:

    /build/macaulay2-1.9.2-10-g06c3063/M2/Macaulay2/c/scc1  -O arithmetic.d
    test -e arithmetic-exports.h && cmp -s arithmetic-exports.h.tmp arithmetic-exports.h || cp arithmetic-exports.h.tmp arithmetic-exports.h
    test -e arithmetic-exports.h && cmp -s arithmetic-exports.h.tmp arithmetic-exports.h || cp arithmetic-exports.h.tmp arithmetic-exports.h
    gcc -g -fdebug-prefix-map=/build/macaulay2-1.9.2-10-g06c3063=. -fstack-protector-strong -Wformat -Werror=format-security -O2 -Wno-unused -pipe -Wall -Wno-shadow -Wcast-qual -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -Wwrite-strings  -Wimplicit-function-declaration -Wno-uninitialized -Wno-cast-qual -Wno-unused -Werror -Wno-unused-label -Wreturn-type -Werror -I. -I./../e -I. -I./../c -I./../regex -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/usr-host/include  -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/usr-host/include  -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/include -I/build/macaulay2-1.9.2-10-g06c3063/M2/usr-host/include  -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/src/gtest -I/usr/local/include/libxml2 -I/usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/cddlib -I/usr/include/cdd  -I/usr/src/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/src/gtest -DNDEBUG -I/usr/local/include/libxml2 -I/usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/cddlib -I/usr/include/cdd  -I/usr/src/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/src/gtest -DNDEBUG -I/usr/local/include/libxml2 -I/usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/x86_64-linux-gnu/singular -I/usr/include/singular -I/usr/include/cddlib -I/usr/include/cdd  -I/usr/src/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/src/gtest -DNDEBUG -DNDEBUG -I/usr/include -I/usr/include -D__FFLASFFPACK_HAVE_CBLAS -g -O2 -fdebug-prefix-map=/build/fflas-ffpack-dZEHwx/fflas-ffpack-2.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -fno-strict-aliasing -fabi-version=6 -std=gnu++11 -fopenmp -Wno-fatal-errors  -c  -Wno-uninitialized -Wno-cast-qual -Wno-unused arithmetic-tmp.c -o arithmetic.o
    cc1: error: command line option '-std=gnu++11' is valid for C++/ObjC++ but not for C [-Werror]
    cc1: all warnings being treated as errors
    ../../include/config.Makefile:195: recipe for target 'arithmetic.o' failed
    make[4]: *** [arithmetic.o] Error 1

Any ideas on how to get around this?

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -222,6 +222,7 @@ AC_SUBST(NTL_WIZARD,no) AC_ARG_ENABLE(ntl-wizard, AS_HELP_STRING(--enable-ntl-wi
 
 AC_SUBST(M2_CPPFLAGS,)
 AC_SUBST(M2_CFLAGS,)
+AC_SUBST(M2_CXXFLAGS,)
 
 AC_PROG_CC() 			# set CFLAGS before this
 AC_SUBST(GCC)			# gets set to yes or no by AC_PROG_CC
@@ -1276,10 +1277,10 @@ test $BUILD_givaro = yes && BUILTLIBS="-lgivaro $BUILTLIBS"
 # test for fflas_ffpack
 if test $BUILD_fflas_ffpack = no
 then AC_MSG_CHECKING(for fflas_ffpack library)
-     FFLAS_FFPACK_CPPFLAGS=`fflas-ffpack-config --cflags 2>&1`
+     FFLAS_FFPACK_CXXFLAGS=`fflas-ffpack-config --cflags 2>&1`
      if test $? = 0
      then AC_MSG_RESULT(found)
-          M2_CPPFLAGS="$M2_CPPFLAGS $FFLAS_FFPACK_CPPFLAGS"
+          M2_CXXFLAGS="$M2_CXXFLAGS $FFLAS_FFPACK_CXXFLAGS"
      else AC_MSG_RESULT(not found, will build)
           BUILD_fflas_ffpack=yes
      fi

--- a/M2/include/config.Makefile.in
+++ b/M2/include/config.Makefile.in
@@ -144,6 +144,7 @@ M2_CFLAGS   += $(M2_BOTH) -Wwrite-strings
 M2_CXXFLAGS += $(M2_BOTH) -Wconversion
 
 M2_CFLAGS += @M2_CFLAGS@
+M2_CXXFLAGS += @M2_CXXFLAGS@
 
 # M2_CFLAGS   += -Werror=implicit-function-declaration -Werror=write-strings $(M2_BOTH)
 ifeq "@__INTEL_COMPILER@" "no"

